pyanaconda/anaconda.py

```diff
@@ -226,6 +242,53 @@ def log_display_mode(self):
                  constants.INTERACTIVE_MODE_NAME[self.interactive_mode],
                  constants.DISPLAY_MODE_NAME[self.display_mode])
 
+    def load_opts(self):
+        """Load argument to variables from self.opts."""
+        self.decorated = self.opts.decorated
+        self.proxy = self.opts.proxy
+        self.updateSrc = self.opts.updateSrc
+        self.methodstr = self.opts.method
+        self.stage2 = self.opts.stage2
+        self.additional_repos = self.opts.addRepo
+
+    def add_additional_repositories_to_ksdata(self):
+        from pyanaconda.kickstart import RepoData
+
+        for add_repo in self.additional_repos:
+            name, repo_url = self._split_additional_repo(add_repo)
+            try:
+                source = SourceFactory.parse_cmdline(repo_url)
+            except PayloadSourceTypeUnrecognized:
+                log.error("Type for additional repository %s is not recognized!", add_repo)
+                return
+
+            repo = RepoData(name=name, baseurl=repo_url, install=False)
+
+            if source.is_nfs or source.is_http or source.is_https or source.is_ftp \
+                    or source.is_file:
+                repo.enabled = True
+            elif source.is_harddrive:
+                repo.enabled = True
+                repo.partition = source.partition
+                repo.iso_path = source.path
+                repo.baseurl = "file://"
+            else:
+                log.error("Source type %s for additional repository %s is not supported!",
+                          source.source_type.value, add_repo)
+                continue
+
+            self._check_repo_name_unique(name)
```

Comment:
Just check `if repo in self.ksdata.repo.dataList()` just like pykickstart does.

Reply:
Fixed.